### ISSUE TITLE
formatting of scale range can be improved

### ISSUE THREAD

**Issue by m-jahn** [showteeth/ggcoverage#17]
Hi!

I noticed that the formatting of the scale range is somewhat inconsistent:

- problem 1: the scale does not work with negative numbers (shows `[0,0]`)
- problem 2: the formatting of numbers by `CeilingNumber()` sometimes creates incorrect numbers

Here's an example, e.g. if one would like to draw the minus strand in negative range:

```
library(ggcoverage)
library(ggplot2)

# import track data
meta.file <- system.file("extdata", "RNA-seq", "meta_info.csv", package = "ggcoverage")
sample.meta <- utils::read.csv(meta.file)
track.folder <- system.file("extdata", "RNA-seq", package = "ggcoverage")

track.df <- LoadTrackFile(
  track.folder = track.folder, format = "bw",
  meta.info = sample.meta
)

# make some non-zero, as in minus strand
track.df[track.df$Type == "KO_rep2", "score"] <- with(track.df, score[Type == "KO_rep2"] * -1)
track.df[track.df$Type == "WT_rep2", "score"] <- with(track.df, score[Type == "WT_rep2"] * -1)

# plot tracks with coloring by 'Group' variable
ggplot() +
geom_coverage(data = track.df, facet.key = "Type", group.key = "Group")
```

![image](https://github.com/showteeth/ggcoverage/assets/32667667/b640be2d-cb2f-4bf6-8530-3158c4259bbd)

This plot also shows the second problem, one of the numbers is formatted as `1250.01`, but in fact it's `1250`.

The output from `CeilingNumber()` is somewhat unpredictable. Note the 0.01 added for some but not all values.

```
> sapply(c(0:10, 100, 123.45, 123.4567), CeilingNumber)
[1]   0.00   1.00   2.00   3.00   4.00   5.00   6.00   7.00   8.01   9.01  10.01 100.00 123.46 123.46
```

I suggest to make some minor changes such that plotting negative values is allowed and also formatting of numbers is simplified. Is there any reason for adding this bit, that I am not aware of? I could add a commit to the current PR or make a new one.

**Comment by m-jahn** [comment]
Suggestion for fomatting the scale numbers:

```
CeilingNumber <- function(x, digits = 2) {
  if (x == 0) {
    "0"
  } else if (abs(x) >= 10^6) {
    formatC(x, format = "e", digits = 2)
  } else if (abs(x) >= 10000) {
    formatC(round(x), format = "f", digits = 0)
  } else if ((x %% floor(x)) != 0) {
    formatC(x, format = "f", digits = 2)
  } else {
    formatC(x, format = "f", digits = 0)
  }
}
```

This makes:

```
> sapply(c(0, 10, 100, 123.45, 123.456789, 1234.6788, 15035.989, 12340234234324), CeilingNumber)
[1] "0"        "10"       "100"      "123.45"   "123.46"   "1234.68"  "15036"    "1.23e+13"
```

I also made a fix to plot negative ranges, it's quite a small change.
I would add this to the PR, OK?

**Comment by showteeth** [comment]
Sorry for the late response, I'm quite busy these days.

Please feel free to create PR, I will check and merge it when I'm free.

Thanks for your suggestions!

YB

**Comment by m-jahn** [comment]
It's fine! Issue was fixed the latest PR, commits [845b016](https://github.com/showteeth/ggcoverage/pull/16/commits/845b016fcc7d4f258fdb0ad72d3543f7cfe93913) and [b432392](https://github.com/showteeth/ggcoverage/pull/16/commits/b4323922cc53dc690849a793d96ce583321bfedc).
